### PR TITLE
v1.13.01

### PR DESCRIPTION
**Pull request by Sherokiel**
WELCOME

### CODE REVIEW

File: Calculator.php [hunk 1]
```diff
@@ -15,8 +15,9 @@
     $history = json_decode($history, true);
 }
 
+welcome(' Wellcome to the calculataor app!','Print "help" to learn more about the app');
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
welcome('Wellcome to the calculataor app!', 'Print "help" to learn more about the app');
```

File: libraries/helpers.php [hunk 2]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
     }
 
     return $result;
-}
\ No newline at end of file
+}
+
+function welcome($msg1,$msg2)
```
Comment (DenTray) [inline libraries/helpers.php]: also let's move it to the console_helpers library

```suggestion
function info_box(...$lines)
```

File: libraries/helpers.php [hunk 3]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+{
+    $length = 45;
```
Comment (DenTray) [inline libraries/helpers.php]: let's get lenght as a max lenght of the one of passed line

File: libraries/helpers.php [hunk 4]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+    $string = 0;
+    $maxStrings = 8;
```
Comment (DenTray) [inline libraries/helpers.php]:
```suggestion
    $maxStrings = count($lines);
```

File: libraries/helpers.php [hunk 5]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+
+    while ($string <= $maxStrings) {
+
```
Comment (DenTray) [inline libraries/helpers.php]:
```suggestion
```

File: libraries/helpers.php [hunk 6]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+        if ($string < 1 || $string > $maxStrings - 1) {
+            $i = 0;
+
+            while ($i < $length) {
+                echo "*";
+                $i++;
+            }
```
Comment (DenTray) [inline libraries/helpers.php]: please use pad str for it

File: libraries/helpers.php [hunk 7]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+            echo PHP_EOL;
+
+        } else {
+            switch ($string)
+            {
```
Comment (DenTray) [inline libraries/helpers.php]:
```suggestion
            switch ($string) {
```

File: libraries/helpers.php [hunk 8]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+                case ($string == round($maxStrings / 4)):
+                    $msg = $msg1;
+                    break;
```
Comment (DenTray) [inline libraries/helpers.php]:
```suggestion

                    break;
```

File: libraries/helpers.php [hunk 9]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+                case ($string == round($maxStrings / 2)):
+                    $msg = $msg2;
+                    break;
```
Comment (DenTray) [inline libraries/helpers.php]:
```suggestion

                    break;
```

File: libraries/helpers.php [hunk 10]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+                default:
+                    $msg = '';
+            }
+            echo "*";
+            echo str_pad($msg, $length - 2, ' ', STR_PAD_BOTH);
+            echo "*" . PHP_EOL;
```
Comment (DenTray) [inline libraries/helpers.php]: generate string to the variable and use info helper to show it

File: libraries/helpers.php [hunk 11]
```diff
@@ -9,4 +9,41 @@ function array_group($array, $key)
...
+        }
+        $string++;
```
Comment (DenTray) [inline libraries/helpers.php]:
```suggestion

        $string++;
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Comment by netherpyro** [comment]
My congratulations for the 21st PR 🎉

**Comment by Sherokiel** [comment]
Хаха))  спасибо

File: Calculator.php [hunk 12]
```diff
@@ -50,22 +51,26 @@
 
 function calculate($argument1, $command, $argument2)
 {
-    switch ($command)
-    {
+    switch ($command) {
         case '+':
             $result = $argument1 + $argument2;
+
             break;
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
            return $argument1 + $argument2;
```

File: libraries/console_helpers.php [hunk 13]
```diff
@@ -31,3 +31,22 @@ function info($result, $emptyLinesCount = 2)
 {
     echo $result . str_repeat(PHP_EOL, $emptyLinesCount);
 }
+
+function info_box(...$lines)
+{
+    foreach ($lines as $maxLines) {
+        $h1 = strlen($maxLines);
+        $lineValue[] = $h1;
+    }
```
Comment (DenTray) [inline libraries/console_helpers.php]:
```suggestion
    $lineLengths = array_map(function () {
        return strlen($maxLines);
    }, $lines);
```

File: libraries/console_helpers.php [hunk 14]
```diff
@@ -31,3 +31,22 @@ function info($result, $emptyLinesCount = 2)
...
+
+    $length = max($lineValue) + 6;
+
+    info (str_pad('*', $length, '*'), 1);
```
Comment (DenTray) [inline libraries/console_helpers.php]:
```suggestion
    info(str_repeat('*', $length), 1);
```

Comment (DenTray) [reply libraries/console_helpers.php]: let's also move it to the separate console helper `write_symbol_line($length, $sybmol)`

**Review by DenTray — CHANGES_REQUESTED** [review]

File: Calculator.php [hunk 15]
```diff
@@ -50,28 +51,29 @@
 
 function calculate($argument1, $command, $argument2)
 {
-    switch ($command)
-    {
+    switch ($command) {
         case '+':
-            $result = $argument1 + $argument2;
-            break;
+
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
```

File: Calculator.php [hunk 16]
```diff
@@ -50,28 +51,29 @@
...
+            return pow($argument1, $argument2);
         case 'sr':
-            $result = pow($argument1, (1 / $argument2));
-    }
 
-    return $result;
+            return pow($argument1, (1 / $argument2));
+        default:
+
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
```

File: Calculator.php [hunk 17]
```diff
@@ -159,7 +161,6 @@ function show_history($history)
 
         foreach ($historyGroups as $date => $historyItems) {
             info ("{$date}:");
-
```
Comment (DenTray) [inline Calculator.php]: please revert empty line

File: Calculator.php [hunk 18]
```diff
@@ -169,7 +170,6 @@ function show_history($history)
 
                 info($historyFunction, 1);
             }
-
```
Comment (DenTray) [inline Calculator.php]: please revert empty line

File: libraries/console_helpers.php [hunk 19]
```diff
@@ -31,3 +31,26 @@ function info($result, $emptyLinesCount = 2)
 {
     echo $result . str_repeat(PHP_EOL, $emptyLinesCount);
 }
+
+function info_box(...$lines)
+{
+    $lineLengths = array_map(function ($maxLines) {
```
Comment (DenTray) [inline libraries/console_helpers.php]:
```suggestion
    $lineLengths = array_map(function ($line) {
```

File: libraries/console_helpers.php [hunk 20]
```diff
@@ -31,3 +31,26 @@ function info($result, $emptyLinesCount = 2)
...
+        return strlen($maxLines);
+    }, $lines);
+
+    $length = max($lineLengths) + 6;
```
Comment (DenTray) [inline libraries/console_helpers.php]: let's move 6 to the variable, defining in the start of the function

File: libraries/console_helpers.php [hunk 21]
```diff
@@ -31,3 +31,26 @@ function info($result, $emptyLinesCount = 2)
...
+
+    write_symbol_line($length, '*');
+
+    foreach ($lines as $msg) {
```
Comment (DenTray) [inline libraries/console_helpers.php]:
```suggestion
    foreach ($lines as $line) {
```

File: libraries/console_helpers.php [hunk 22]
```diff
@@ -31,3 +31,26 @@ function info($result, $emptyLinesCount = 2)
...
+        $message = str_pad($msg, $length - 6, ' ', STR_PAD_BOTH);
+
+        info("*  {$message}  *", 1);
+    }
+    write_symbol_line($length, '*');
```
Comment (DenTray) [inline libraries/console_helpers.php:52]:
```suggestion

    write_symbol_line($length, '*');
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**